### ISSUE TITLE
Groups parameter of Conv2d and Conv2dTranpose ('deconvolution') not Working ?

### ISSUE THREAD

**Issue by NguyenThaiHoc1** [tensorflow/tensorflow#45216]
Hello Authors

I'm a user of tensorflow. Someday i have use Deconvolution and i know it called "Conv2dTranpose" in tensorflow. But I want to reduce parameters of model by groups. I found groups property at Pytorch and I see it in Conv2D class (Conv2dTranspose inheritance Conv2d). But when i use i get result which i don't want 
This is my code 

`
import tensorflow as tf
from tensorflow.keras.layers import Conv2D, Conv2DTranspose, BatchNormalization, ReLU, MaxPool2D
model = tf.keras.Sequential([
    Conv2D(512,
           kernel_size=(3, 3),
           strides=(1, 1),
           padding='same',
           use_bias=False,
           name='conv'),
    BatchNormalization(),
    ReLU(),
    Conv2DTranspose(256, kernel_size=(4, 4), strides=(2, 2),
                    padding='same', use_bias=False,
                    groups=256,
                    kernel_initializer='he_normal'),
    BatchNormalization(),
    ReLU()
])

model.build((32, 256, 192, 3))
model.summary()
`

This is summary when i received 
>
Layer (type)                 Output Shape              Param #   
conv (Conv2D)                (32, 256, 192, 512)       13824     
batch_normalization (BatchNo (32, 256, 192, 512)       2048      
re_lu (ReLU)                 (32, 256, 192, 512)       0         
conv2d_transpose (Conv2DTran (32, 512, 384, 256)       2097152   
batch_normalization_1 (Batch (32, 512, 384, 256)       1024      
re_lu_1 (ReLU)               (32, 512, 384, 256)       0         
Total params: 2,114,048
Trainable params: 2,112,512
Non-trainable params: 1,536

I think conv2d_transpose will: 2097152 / 256(groups) = 8192 params ? 

Sorry for my writer is not good 
Thanks you for reading 
Thai Hoc

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/d8410424a44ce181b1e9fca6bd70e1fe/45216.ipynb). Thanks!

**Comment by aaronmondal** [comment]
Indeed, I get
```python
>>> conv_t = Conv2DTranspose(32, 3, 1, groups=32)
>>> conv_t(tf.ones([1, 64, 64, 32]))
>>> print(conv_t.kernel.shape)
(3, 3, 32, 32)
```
while
```python
>>> conv = Conv2D(32, 3, 1, groups=32)
>>> conv(tf.ones([1, 64, 64, 32]))
>>> print(conv_t.kernel.shape)
(3, 3, 1, 32)
```
You could try this, however:
```python
def grouped_conv2d_transpose(inputs, filters, kernel_size, strides, groups):
    """Performs grouped transposed convolution.

    Args:
        inputs: A `Tensor` of shape `[batch_size, h, w, c]`.
        filters: The number of convolutional filters.
        kernel_size: The spatial size of the convolutional kernel.
        strides: The convolutional stride.
        groups: The number of groups to use in the grouped convolution step.
            The input channel count needs to be evenly divisible by `groups`.
    Returns:
        A `Tensor` of shape `[batch_size, new_h, new_w, filters]`.
    """
    splits = tf.split(inputs, groups, axis=-1)
    convolved_splits = [
        tf.keras.layers.Conv2DTranspose(
            filters,
            kernel_size,
            strides
        )(split) for split in splits
    ]
    return tf.concat(convolved_splits, -1)
```

**Comment by rmothukuru** [comment]
@NguyenThaiHoc1,
Can you please let us know if the above workaround worked for you? Thanks!

**Comment by rmothukuru** [comment]
@NguyenThaiHoc1,
Can you please respond to the above comment. Thanks!

**Comment by aaronmondal** [comment]
Note that the workaround will be considerably slower on GPUs than regular grouped `conv2d` since it does not make use of the grouped CUDA kernel. Maybe this should be flagged as a bug.

**Comment by bhack** [comment]
Can we set this as `contribution welcome`?

**Comment by mohantym** [comment]
I was able to replicate this issue in TF 2.7 .Attaching [Gist](https://colab.research.google.com/gist/mohantym/9434e0602d92b1167766ed3940b4599c/45216.ipynb#scrollTo=ceJRIVOPfZVJ) for reference. Thanks!

**Comment by manish-p-gupta** [comment]
Hey is this issue being worked on? If not then I'd like to give it a try.

**Comment by ElectricalMichael** [comment]
As of now, this issue is still not fixed, correct? 
In my case, I want to deploy the model on an FPGA and therefore need to export it to a valid .h5 file. For this (and the quantization step), I need to execute the model with correct behaviour regarding the groups arguments, therefore the above solution does not work for me.

Is there any progress? @mohantym @manish-p-gupta

**Comment by NickProgramming** [comment]
Hi,

I had the same problem, but fixed it myself by changing the kernel shape creation in the Conv2dTranspose Class. In the build function the kernel shape is getting calculated. When comparing to the normal Conv2d I think the following line in the build function should be changed:

/keras/layers/convolutional)/conv2d_transpose.py
`kernel_shape = self.kernel_size + (self.filters, input_dim)`
to
`kernel_shape = self.kernel_size + (self.filters // self.groups, input_dim)`

**Comment by ho0-kim** [comment]
I struggled with this issue on _Apple M1_ machine.
_groups_ parameter doesn't work on _Apple Silicon_, but it works on other machine well.
The tested tensorflow version is 2.13.0-rc1

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
